ui/src/Auth/TokenProvider.tsx

```diff
@@ -16,8 +16,30 @@
  */
 
 import Cookies from 'universal-cookie';
+import jwtDecoder from 'jwt-decode';
 
 export const getToken = (): string => {
     const cookies = new Cookies();
     return cookies.get('accessToken');
 };
+
+export const removeToken = (): void => {
+    const cookie = new Cookies();
+    cookie.remove('accessToken', {path: '/'});
+};
+
+export const getDecodedToken = (): unknown | null  => {
+    const accessToken = getToken();
+    return jwtDecoder(accessToken);
+};
+
+export const getUserNameFromAccessToken = (): string => {
+    try {
+        const decodedAccessToken = getDecodedToken();
+        // @ts-ignore
+        const userName = decodedAccessToken.sub;
+        return userName;
+    } catch {
+        return '';
```

Comment:
Will we wind up with "Welcome, " if jwtdecoder doesn't work? Should we remove the comma if there's no name?

Reply:
There's a check in the html so it won't actually show up.